### ISSUE TITLE
The alternativeTitlesDomain.rng declares the 'title-role' attribute as required

### ISSUE THREAD

**Issue by takesson** [oasis-tcs/dita#915]
The alternativeTitlesDomain.rng declares the 'title-role' attribute as required and with a default value. This has 2 negative side-effects:
 - Editors (e.g. Oxygen) might add the 'title-role' attribute when creating an element like 'titlehint'. That's typically not necessary nor desirable.
 - Trang will omit the default value when converting RNG to XSD (perhaps also to DTD). The attribute will be required but without the default value.

Declaring title-role attribute as required is inconsistent with most other attributes with a default value. 

Thanks!

**Comment by raducoravu** [comment]
> Editors (e.g. Oxygen) might add the 'title-role' attribute when creating an element like 'titlehint'. That's typically not necessary nor desirable.

I tested and Oxygen indeed inserts the attribute like this:

     <navtitle title-role="navigation"></navtitle>

Oxygen would not insert in general the default value of an attribute but in this case the attribute is considered required. I have not seen many situations where an attribute is considered both required and also has a default value in the schema, usually it's either one or the other.
For example if the attribute would be defined as optional in the RNG, then Oxygen would no longer feel compelled to insert it:

            <define name="navtitle.attributes">
                <optional>
                    <attribute name="title-role" a:defaultValue="navigation"/>
                </optional>

**Comment by robander** [comment]
We should discuss this at the TC -- per the latest spec, the attribute is indeed required. Looking at Jarno's handy rendering of the latest spec, it's described here:
https://dita-lang.org/dita/langref/attributes/commonattributes#common-atts__attr-title-role

edit: if I'm understanding right, the issue is that when there is a default value, the RNG should not also mark it required -- it's used regardles, so this is just a grammar quirk with no real impact other than tool behaviors like the one in Oxygen.

**Comment by takesson** [comment]
> edit: if I'm understanding right, the issue is that when there is a default value, the RNG should not also mark it required -- it's used regardles, so this is just a grammar quirk with no real impact other than tool behaviors like the one in Oxygen.

Correct. RNG defines default values separately from whether its required. The DTD definition of "title-role" attributes is actually not strictly clear either because it only states a default value instead of explicitly stating #REQUIRED and a default value. Based on Microsoft's DTD reference I would consider the DTD specification a bit vague.

It seems like a required attribute with default value is more or less unprecedented in the DITA specification. Correct? I am unable to find any other such attribute in RNG and no uses of #REQUIRED combined with default value in the DTDs.

These attributes are all declared optional in the RNG: class, placement, linking (glossref), toc, search, keycol (choicetable), ...

The SVG specification is also declaring optional attributes with default value in the RNG.
 
https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/ms256140(v=vs.100)
